lib/web/desktop_playback.go

```diff
@@ -170,10 +170,22 @@ func (pp *playbackPlayer) streamSessionEvents(ctx context.Context, cancel contex
 			if !errors.Is(err, context.Canceled) {
 				pp.log.WithError(err).Errorf("streaming session %v", pp.sID)
 			}
+
 			return
 		case evt := <-eventsC:
 			if evt == nil {
 				pp.log.Debug("reached end of playback")
+
+				msg, err := utils.FastMarshal(struct {
+					Message string `json:"message"`
+				}{Message: "end"})
+				if err != nil {
+					pp.log.WithError(err).Errorf("failed to marshal end message into JSON: %v", msg)
+				}
+				if _, err := pp.ws.Write(msg); err != nil {
```

Comment:
If the marshal failed, this will attempt to write an empty message to the websocket. Doesn't seem like that's what you want.